mllib/src/main/scala/org/apache/spark/mllib/clustering/GaussianMixture.scala

```diff
@@ -168,16 +182,26 @@ class GaussianMixture private (
       // Create new distributions based on the partial assignments
       // (often referred to as the "M" step in literature)
       val sumWeights = sums.weights.sum
-      var i = 0
-      while (i < k) {
-        val mu = sums.means(i) / sums.weights(i)
-        BLAS.syr(-sums.weights(i), Vectors.fromBreeze(mu),
-          Matrices.fromBreeze(sums.sigmas(i)).asInstanceOf[DenseMatrix])
-        weights(i) = sums.weights(i) / sumWeights
-        gaussians(i) = new MultivariateGaussian(mu, sums.sigmas(i) / sums.weights(i))
-        i = i + 1
+      if (distributeGaussian) {
+        sc.parallelize(0 until k).map{ i =>
+          val mu = sums.means(i) / sums.weights(i)
+          BLAS.syr(-sums.weights(i), Vectors.fromBreeze(mu),
+            Matrices.fromBreeze(sums.sigmas(i)).asInstanceOf[DenseMatrix])
+          weights(i) = sums.weights(i) / sumWeights
```

Comment:
Doing this does not get the updated weights/gaussians back to the driver (ie, the weights/gaussians arrays operated on here are local to the node running the task, only).  Try something like the code I posted on Jira for this.